codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -384,7 +383,7 @@ public boolean merge(ChannelHandlerContext ctx, Http2RemoteFlowController.FlowCo
         private final short weight;
         private final boolean exclusive;
 
-        private FlowControlledHeaders(Http2Stream stream, Http2Headers headers, int streamDependency, short weight,
+        public FlowControlledHeaders(Http2Stream stream, Http2Headers headers, int streamDependency, short weight,
```

Comment:
I would prefer to set the inner access modifiers independently of the outer scope (unless there is good reason not to).  I don't have strong objection to this being package-private, but I will leave as is because it is already cherry-picked.